learn/getting_started/algolia_migration.md

```diff
@@ -0,0 +1,249 @@
+
+```bash
+npm install algoliasearch meilisearch
+```
+
+### Create Algolia client
+
+You'll need your **Application ID** and **API Key** to start the Algolia client. Both can be found in your [Algolia account](https://www.algolia.com/api-keys).
+
+Paste the below code in `script.js`:
+
+```js
+const AlgoliaSearch = require("algoliasearch");
+const algoliaClient = AlgoliaSearch(
+  "APPLICATION_ID",
+  "ADMIN_API_KEY"
+);
+const algoliaIndex = algoliaClient.initIndex("INDEX_NAME");
+```
+
+Replace `APPLICATION_ID` and `ADMIN_API_KEY` with your Algolia application ID and admin API key respectively.
+
+Replace `INDEX_NAME` with the name of the Algolia index you would like to migrate to Meilisearch.
+
+### Fetch data from Algolia
+
+To fetch all Algolia index data at once, use Algolia's [`browseObjects`](https://www.algolia.com/doc/api-reference/api-methods/browse/) method.
+
+```js
+let records = [];
+await algoliaIndex.browseObjects({
+    batch: (hits) => {
+      records=records.concat(hits)
+    }
+  });
+```
+
+On each batch of hits, the `batch` callback method is invoked, and the content is concatenated in the `records` array. This variable will be used later on in the upload process.
+
+#### Nested objects
+
+Meilisearch versions prior to v0.27.0 have limited support for nested fields, meaning that **nested objects must be flattened manually**. This can be avoided entirely by using [the latest version of Meilisearch](https://github.com/meilisearch/MeiliSearch/releases). If this is impossible for some reason, try the modified data retrieval process below.
+
+Let's say your document structure looks like this:
+
+```json
+{
+    "objectID": 1,
+    "name": "John",
+    "age": 20,
+    "address": {
+        "country": "Spain"
+    }
+}
+```
+
+To export this data from Algolia and simultaneously flatten it, use the same `browseObjects` method as described above. However, before concatenation, flatten the `address` object into `address.country`, and perform the same operation for other subfields:
+
+```js
+let records = [];
+await algoliaIndex.browseObjects({
+ batch: (recordsArr) => {
+        records = recordsArr.reduce((acc, record) => {
+          const { address, ...rest } = record;
+          acc.push({ "address.country": address.country, ...rest });
+          return acc;
+        }, records);
+      }
+});
+```
+
+Adapt this principle to your own dataset by replacing `address` and `address.country` with a parent object and nested field, respectively.
+
+### Create Meilisearch client
+
+Create a Meilisearch client by passing the host URL and API key of your Meilisearch instance. The easiest option is to use the automatically generated [admin API key](/learn/security/master_api_keys.md#listing-api-keys).
+
+```js
+const { MeiliSearch } = require("meilisearch");
+const meiliClient = new MeiliSearch({
+  host: "MEILI_HOST",
+  apiKey: "MEILI_API_KEY",
+});
```

Comment:
Just for reference, I mentioned capitalisation as a JS convention to indicate a function should be invoked with `new` and how `const AlgoliaSearch` broke that convention. 

Here you see that the constant `Meilisearch` is capitalised and then invoked as `new Meilisearch ([etc])`, which is what I would expect as a JS programmer.